frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx

```diff
@@ -236,26 +241,36 @@ function DashCardInner({
     showClickBehaviorSidebar(dashcard.id);
   }, [dashcard.id, showClickBehaviorSidebar]);
 
-  const changeCardAndRunHandler = useMemo(() => {
-    if (!navigateToNewCardFromDashboard) {
+  const changeCardAndRunHandler =
+    useMemo<DashCardOnChangeCardAndRunHandler | null>(() => {
+      if (!navigateToNewCardFromDashboard) {
+        return null;
+      }
+
+      return ({ nextCard, previousCard, objectId }) => {
```

Comment:
honestly it looks like a `useCallback` + conditional passing handler: `navigateToNewCardFromDashboard ? changeCardAndRunHandler: null)`

Reply:
18709164627f6f94cb721e01ac818dc295b7729f